packages/bull-vault/src/AuctionBull.sol

```diff
@@ -750,26 +709,26 @@ contract AuctionBull is UniFlash, Ownable, EIP712 {
     function _executeLeverageComponentRebalancing(
         ExecuteLeverageComponentRebalancingParams memory _params
     ) internal {
-        uint256 remainingWeth = IERC20(weth).balanceOf(address(this));
+        // uint256 remainingWeth = IERC20(weth).balanceOf(address(this));
         uint256 wethInCollateral = IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy));
-        if (_params.wethTargetInEuler > remainingWeth.add(wethInCollateral)) {
+        if (_params.wethTargetInEuler > _params.wethSoldInAuction.add(wethInCollateral)) {
             // have less ETH than we need in Euler, we have to buy and deposit it
             // borrow more USDC to buy WETH
-            uint256 wethToBuy = _params.wethTargetInEuler.sub(remainingWeth.add(wethInCollateral));
+            uint256 wethToBuy = _params.wethTargetInEuler.sub(_params.wethSoldInAuction.add(wethInCollateral));
             _exactOutFlashSwap(
                 usdc,
                 weth,
                 _params.ethUsdcPoolFee,
                 wethToBuy,
                 wethToBuy.wmul(_params.wethLimitPrice).div(WETH_DECIMALS_DIFF),
                 uint8(FLASH_SOURCE.FULL_REBALANCE_BORROW_USDC_BUY_WETH),
-                abi.encodePacked(wethToBuy.add(remainingWeth))
+                abi.encodePacked(wethToBuy.add(_params.wethSoldInAuction))
             );
         } else {
             // have more ETH than we need in either Euler or from withdrawing from crab
             //we need to sell ETH and either deposit or withdraw from euler
-            uint256 wethToSell = remainingWeth.add(wethInCollateral).sub(_params.wethTargetInEuler);
-            // wethToSell + wEthTargetInEuler = remainingWeth+wethInCollateral
+            uint256 wethToSell = _params.wethSoldInAuction.add(wethInCollateral).sub(_params.wethTargetInEuler);
+            // wethToSell + wEthTargetInEuler = _params.wethSoldInAuction+wethInCollateral
```

Comment:
commentz

Reply:
I thought they are there explicitly for more explanation